lifecycle/services/export.go

```diff
@@ -0,0 +1,174 @@
+	return ""
+}
+
+func (rbs *ReleaseBundlesService) ExportReleaseBundle(rlExportParams *ReleaseBundleExportParams, queryParams CommonOptionalQueryParams) (exportResponse ReleaseBundleExportedStatusResponse, err error) {
+	// Check the current status
+	if exportResponse, err = rbs.getExportedReleaseBundleStatus(rlExportParams, queryParams); err != nil {
+		return
+	}
+	// Trigger export if needed
+	if exportResponse.Status == ExportNotTriggered || exportResponse.Status == ExportFailed {
+		if err = rbs.triggerReleaseBundleExportProcess(rlExportParams, queryParams); err != nil {
+			return
+		}
+	}
+	// Wait for export to finish
+	if exportResponse.Status != ExportCompleted {
+		if exportResponse, err = rbs.checkExportedStatusWithRetries(rlExportParams, queryParams); err != nil {
+			return
+		}
+	}
+	return
+}
+
+func (rbs *ReleaseBundlesService) checkExportedStatusWithRetries(rlExportParams *ReleaseBundleExportParams, queryParams CommonOptionalQueryParams) (response ReleaseBundleExportedStatusResponse, err error) {
+	pollingAction := func() (shouldStop bool, responseBody []byte, err error) {
+		response, err = rbs.getExportedReleaseBundleStatus(rlExportParams, queryParams)
+		if err != nil {
+			return
+		}
+		if err != nil {
+			return true, nil, err
+		}
+		switch response.Status {
+		case ExportProcessing:
+			return false, nil, nil
+		case ExportCompleted, ExportFailed:
+			return true, responseBody, nil
+		default:
+			return true, nil, errorutils.CheckErrorf("received unexpected status: '%s'", response.Status)
+		}
+	}
+	pollingExecutor := &httputils.PollingExecutor{
+		Timeout:         defaultMaxWait,
+		PollingInterval: SyncSleepInterval,
+		PollingAction:   pollingAction,
+		MsgPrefix:       fmt.Sprintf("Getting Exported Release Bundle %s/%s status...", rlExportParams.ReleaseBundleName, rlExportParams.ReleaseBundleVersion),
+	}
+	_, err = pollingExecutor.Execute()
+	return
+}
+
+func (rbs *ReleaseBundlesService) triggerReleaseBundleExportProcess(rlExportParams *ReleaseBundleExportParams, queryParams CommonOptionalQueryParams) (err error) {
+	operation := &exportOperation{
+		reqBody:     RbExportBody{*rlExportParams},
+		queryParams: queryParams,
+	}
+	log.Debug("Triggering Release Bundle Export process...")
+	_, err = rbs.doOperation(operation)
+	return
+}
+
+func (rbs *ReleaseBundlesService) getExportedReleaseBundleStatus(rlExportParams *ReleaseBundleExportParams, queryParams CommonOptionalQueryParams) (exportedStatusResponse ReleaseBundleExportedStatusResponse, err error) {
+	operation := &exportStatusOperation{
+		reqBody:     RbExportBody{*rlExportParams},
+		queryParams: queryParams,
+	}
+	operationExtraParams := operationParams{
+		httpMethod:  http.MethodGet,
+		contentType: "application/json",
+	}
+	log.Debug("Getting Release Bundle Export status...")
+	respBody, err := rbs.doOperation(operation, operationExtraParams)
+	if err != nil {
+		return
+	}
+	err = json.Unmarshal(respBody, &exportedStatusResponse)
```

Comment:
```suggestion
	err = errorutils.CheckError(json.Unmarshal(respBody, &exportedStatusResponse))
```
Cover potential errors where they first appear in our code (generated by us / third party packages) with the `CheckError` function. This function is a debug tool that can be used to identify where an error was thrown from.